topology/nhlt/nhlt-processor.c

```diff
@@ -253,6 +256,15 @@ int nhlt_create(struct tplg_pre_processor *tplg_pp)
 	for (i = 0; i < MAX_ENDPOINT_COUNT; i++)
 		eps[i] = NULL;
 
+	/* we always have only 0 or 1 dmic ep */
+	if (nhlt_dmic_get_ep_count(tplg_pp)) {
+		/* the index is always 0 in dmic case */
+		ret = nhlt_dmic_get_ep(tplg_pp, &eps[eps_count], 0);
+		if (ret < 0)
+			return ret;
```

Comment:
you need to unwind and free previously successful ep alloc's right?

Reply:
there isn't any ep's allocated before this...